### ISSUE TITLE
This is not production ready library

### ISSUE THREAD

**Issue by LukasKubicek** [microsoft/fluentui-blazor#1120]
I am feeling like I wasted two days trying to work with this UI library, fighting issue after issue.. like for example it is either possible to have FluentTextArea in form but not FluentButton for submit, because there is a bug that pressing enter in FluentTextArea fires the submit event.. this is such an important functionality..  

Documentation is very.. minimal.. It took me few hours to create form that looks decent, something that with Bootstrap is just a copy paste from their documentation....

The point of having BLAZOR UI COMPONENTS LIBRARY is that we shouldn't have to touch javascript and css very much. But when you remove FluentFooter from the MainLayout.razor there is dead space on the bottom of the screen because the css file that comes with the project template has hardcoded width-96px. Terrible experience.

The FluentDataGrid has some issues that sometimes breaks content overflow behavior.. when reported, author didn't even want to investigate this, just good enough that it started working eventually, but nobody knows why exactly..

This doesn't feel like production ready UI library at all :( maybe it should just be css skin instead

After two days I am considering to redo everything in some other UI library or pure blazor with Bootstrap..

Sorry for the rant, maybe it will help someone avoid the frustration that I am currently experiencing. It would be nice to be more honest about the state of this UI library on the presentation page.. this is not production ready.

**Comment by vnbaaij** [comment]
Please go redo your work with another library. Hope you will have have more success with it!

I'm **not** giving you a pass ranting like. It is **not** appreciated and does **not** do any right to the countless hours me and my fellow authors have put into this library. You opended 5 issues (including this one) in the last 24 hours. I responded to all of them pretty quickly, I believe, with nothing but respect and trying to be helpful. I'm all for constructive critisism but this is not that. 

Again, please just go.

**Comment by LukasKubicek** [comment]
Thank you for responding quickly and sorry I let my frustration like this.

Wish you best of luck. I only hope that other people do not waste time like I feel I did.

**Comment by dvoituron** [comment]
As Vincent has already indicated, I also find that your message adds nothing constructive to the project we're trying to develop. I understand that you're frustrated by features that don't work as you'd imagined, by bugs that are present in our Lib, as in all software. But to denigrate in this way is more than discourteous.

In any case, we're a **community project** in which you can contribute, just like any other developer.

If you encounter any difficulties or problems, please let us know so that we can improve the library, the documentation and the examples that are and will be offered.

Or even better, create [Pull Requests](https://github.com/microsoft/fluentui-blazor/pulls) to propose or correct these functionalities.

I hope you'll find the best Libs for your developments.

**Comment by LukasKubicek** [comment]
I don't mean to disrespect your work. My mistake was that all what I really needed was css skin over either default blazor components or some mature blazor UI library and I didn't realize this in time. I understand that creating UI components library is a monumental effort. Your competitors have large team of developers that work on similar libraries full time. 

Reality is that this library is nowhere near the commercial alternatives and also your library introduces bugs that are not present in default vanilla blazor components and are pretty critical (e.g. the textarea submit on enter) for production use. Yes, everything can be "fixed" with some workarounds, and maybe that is ok for someone - I just didn't expected this to be the state of this when this is presented as production ready library from Microsoft.

I would just silently redo everything in pure blazor with some css and not bother you again, but if it will help at least one person to avoid this mistake and waste their time I will be glad.

In the end the mistake of wasted time is purely mine.

**Comment by LukasKubicek** [comment]
If I may have one suggestion: on the web page, write something about what this is. So maybe it will help clear some confusion for people like me. I saw fluentui-blazor on the recent dotnet conf (when .net 8 was introduced). I didn't hear about it before and because it looks like it is from Microsoft I assumed that it is just a different look/skin/css for the default Blazor components that comes with the framework. In reality it uses very different underlying html/js/css componentes from some other project. This introduces dependencies, if they have bug, this library has to wait for them to fix this.. I had no idea that this is how fluentui-blazor works. I was a bit surprised that instead of being just a skin over default blazor components everything is using new FluentXXX elements, but didn't think about it much.. I should have.

Then there is this weird dichotomy that this is not just skin, but complete library with elements for everything from layout to inputs, but there is just very minimalistic documentation on places where I would expect how to create e.g. complex forms - since everything  is done with these new FluentXXX elements, why don't they show us more examples how to use this to format inputs etc? I am still a bit confused what is the place/role of the fluentui-blazor.. If you look at proper full UI blazor libraries, they all have examples how to do complex forms with them. You refuse to provide that, but also you are not just a skin..

So I don't know.. maybe few words somewhere about the project and what it is and what it isn't. You have Microsoft name on you - if this was just some opensource project of few guys, expectations would be different - but this is under github.com/microsoft so it couldn't be more official than that.. but the experience is nowhere what I would expect from something production ready from Microsoft..

**Comment by vnbaaij** [comment]
> If I may have one suggestion: on the web page, write something about what this is. So maybe it will help clear some confusion for people like me. I saw fluentui-blazor on the recent dotnet conf (when .net 8 was introduced). I didn't hear about it before and because it looks like it is from Microsoft I assumed that it is just a different look/skin/css for the default Blazor components that comes with the framework. In reality it uses very different underlying html/js/css componentes from some other project. This introduces dependencies, if they have bug, this library has to wait for them to fix this.. I had no idea that this is how fluentui-blazor works. I was a bit surprised that instead of being just a skin over default blazor components everything is using new FluentXXX elements, but didn't think about it much.. I should have.

Literaly the first sentence on the [home page](https://www.fluentui-blazor.net/) of our site: "The Microsoft.FluentUI.AspNetCore.Components package provides a set of Blazor components. Some of the components are wrappers around Microsoft's official Fluent UI Web Components. Others are components that leverage the Fluent Design System or make it easier to work with Fluent".

You said you looked at our .NET Conf talk. We started that talk by explaining what FAST is, what the Fluent UI Web components are and how we are using all that to build this library. You assumed wrong. Don't blame us for that. (You know what they say about assumptions 😉)

> Then there is this weird dichotomy that this is not just skin, but complete library with elements for everything from layout to inputs, but there is just very minimalistic documentation on places where I would expect how to create e.g. complex forms - since everything is done with these new FluentXXX elements, why don't they show us more examples how to use this to format inputs etc? I am still a bit confused what is the place/role of the fluentui-blazor.. If you look at proper full UI blazor libraries, they all have examples how to do complex forms with them. You refuse to provide that, but also you are not just a skin..

The whole demo site is filled with all kinds of examples for all the components we have! We indeed do not have exmples on how to compose a what you call 'complex form', but we do have examples for the building blocks that you can use to  achieving that. We don't 'refuse' to provide something like that but we just don't have it (yet). Excellent opportinity for a contribution, I would think! (...again some disrespect here by calling us a not proper full library)

> So I don't know.. maybe few words somewhere about the project and what it is and what it isn't. You have Microsoft name on you - if this was just some opensource project of few guys, expectations would be different - but this is under github.com/microsoft so it couldn't be more official than that.. but the experience is nowhere what I would expect from something production ready from Microsoft..

It **is** an **open source** project (under active development) from a few guys that happen to work for Microsoft and is published under that repo becuse that is where all Microsoft owned open sourced code resides (i.e. the code is owned by Microsoft and not by the individual developers. Why would you think that makes anything published there different than any other open source project?

**Comment by osmanm786** [comment]
I beg to differ. I have a few clients in prod with this library. This has become my go to library. Yeah I need to tweek things here and there but so what.  But enough of that. I like where the new theme component is going. Great jod! 

Also, these guys grew this library really quickly.  I was kinda wondering if these guys get days off.

**Comment by dajma00** [comment]
I have used MudBlazor, Syncfusion, Bootstrap etc. etc. with Blazor. ALL libraries will have a bug every now and then. So we learn to live with that, find work arounds, raise issues etc. I haven't tried FAST but it may work for some, not for others and that is true of all libraries, frameworks and software in general.

**Comment by AathifMahir** [comment]
In my case, I already have enterprise level application in prod using Fluent Blazor and during the development I had some fare share of issues that I have faced, but I have sorted most of them out. Many issues that I have faced related to myself who didn't look into demo site thoroughly but with all respect trashing the authors for our mistake is little over board, please be kind understand what went wrong before blaming others.

**Comment by dosni** [comment]
I love FluentUI Blazor Component.  Usually when I created User interface, someone has to fix my user interfaces.  But after using FluentUI, they said they are good enough. Well I have to credit fluentUI component for better looks.

Yes it is a bit more difficult compared to another component I used before. For me, it is worth the time to learn how to use it.  Not to mention about the sample code from demo site of fluentUI - so many things I learn from that sample code. Thank you.

**Comment by LukasKubicek** [comment]
If I new all this at the beginning I just wouldn't choose this library. Nothing against small open source projects, on the contrary. I just had no idea that this project is like this. I assumed that being under github.com/microsoft means something else. Now I know.

Btw what I really needed was something like this: https://github.com/mdbootstrap/bootstrap-fluent-design
I switched to vanailla blazor with this css and that is exactly what I need.

Btw2 while rewriting my app I discovered anothe bug - your login page doesn't work when remember me checkbox is checked - testet on new fluent ui blazor server project without any changes.

Feel free to close this issue when you feel like it.
